static/app/components/events/interfaces/crashContent/stackTrace/content.tsx

```diff
@@ -172,240 +180,188 @@ class Content extends Component<Props, State> {
+
+        return maxLengthUntilThisPoint > relativeAddress.length
+          ? maxLengthUntilThisPoint
+          : relativeAddress.length;
+      } catch {
+        return maxLengthUntilThisPoint;
+      }
+    },
+    0
+  );
 
-    const isFrameAfterLastNonApp = this.isFrameAfterLastNonApp();
-    const mechanism =
-      platform === 'java' && event.tags?.find(({key}) => key === 'mechanism')?.value;
-    const isANR = mechanism === 'ANR' || mechanism === 'AppExitInfo';
-
-    (data.frames ?? []).forEach((frame, frameIdx) => {
-      const prevFrame = (data.frames ?? [])[frameIdx - 1];
-      const nextFrame = (data.frames ?? [])[frameIdx + 1];
-      const repeatedFrame = this.isRepeatedFrame(frame, nextFrame);
+  let convertedFrames = frames
+    .map((frame, frameIndex) => {
+      const prevFrame = frames[frameIndex - 1];
+      const nextFrame = frames[frameIndex + 1];
+      const repeatedFrame = isRepeatedFrame(frame, nextFrame);
 
       if (repeatedFrame) {
         nRepeats++;
       }
 
       if (
-        (this.frameIsVisible(frame, nextFrame) && !repeatedFrame) ||
-        hiddenFrameIndices.includes(frameIdx)
+        (frameIsVisible(frame, nextFrame) && !repeatedFrame) ||
+        hiddenFrameIndices.includes(frameIndex)
       ) {
-        const image = this.findImageForAddress(frame.instructionAddr, frame.addrMode);
-        frames.push(
-          <DeprecatedLine
-            key={frameIdx}
-            event={event}
-            data={frame}
-            isExpanded={expandFirstFrame && lastFrameIdx === frameIdx}
-            emptySourceNotation={lastFrameIdx === frameIdx && frameIdx === 0}
-            isOnlyFrame={(data.frames ?? []).length === 1}
-            nextFrame={nextFrame}
-            prevFrame={prevFrame}
-            platform={platform}
-            timesRepeated={nRepeats}
-            showingAbsoluteAddress={showingAbsoluteAddresses}
-            onAddressToggle={this.handleToggleAddresses}
-            image={image}
-            maxLengthOfRelativeAddress={maxLengthOfAllRelativeAddresses}
-            registers={{}} // TODO: Fix registers
-            isFrameAfterLastNonApp={isFrameAfterLastNonApp}
-            includeSystemFrames={includeSystemFrames}
-            onFunctionNameToggle={this.handleToggleFunctionName}
-            onShowFramesToggle={e => {
-              this.handleToggleFrames(e, frameIdx);
-            }}
-            isSubFrame={hiddenFrameIndices.includes(frameIdx)}
-            isShowFramesToggleExpanded={toggleFrameMap[frameIdx]}
-            showCompleteFunctionName={showCompleteFunctionName}
-            isHoverPreviewed={isHoverPreviewed}
-            frameMeta={meta?.frames?.[frameIdx]}
-            registersMeta={meta?.registers}
-            debugFrames={debugFrames}
-            isANR={isANR}
-            threadId={threadId}
-            lockAddress={lockAddress}
-            hiddenFrameCount={frameCountMap[frameIdx]}
-          />
-        );
+        const frameProps = {
```

Comment:
To make the typechecking a bit better, I'd export the props type from DeprecatedLine and add it here like `const frameProps: DeprecatedLineProps`. That'll give better error messages when something goes wrong (for future edits)